esx_service/utils/auth_api.py

```diff
@@ -777,14 +857,29 @@ def check_privilege_parameters(privilege):
     # If both volume max size and volume total size are set,
     # volume max size should not exceed volume total size
     if (volume_maxsize and volume_totalsize and (volume_maxsize > volume_totalsize)):
-        error_info = error_code.generate_error_info(ErrorCode.PRIVILEGE_INVALID_VOLUME_SIZE, volume_maxsize, volume_totalsize)
+        error_info = generate_error_info(ErrorCode.PRIVILEGE_INVALID_VOLUME_SIZE, volume_maxsize, volume_totalsize)
         return error_info
 
     return None
 
+def check_usage_quota(datastore, volume_totalsize_in_MB):
+    """
+        Check the usage_quota to be set in datastore is valid or not
+        Return None if the usage_quota is valid
+        Return error_info if the usage_quota is invalid
+    """
+    # usage_quota on "__VM_DS" and "__ALL_DS" should be unlimited (volume_totalsize_in_MB shoule be 0)
+    if datastore == auth_data_const.VM_DS or datastore == auth_data_const.ALL_DS:
+        if volume_totalsize_in_MB is not None and volume_totalsize_in_MB > 0:
+            err_msg = "Only unlimit is allowed"
```

Comment:
the message is confusing. Also - why I cannot put a limit on _ALL_DS ? This sounds wrong... We may still keep it to merge. At any rate, please elaborate here